src/test/ui/drop/repeat-drop.rs

```diff
@@ -0,0 +1,69 @@
+// run-pass
+
+static mut CHECK: usize = 0;
+
+struct DropChecker(usize);
+
+impl Drop for DropChecker {
+    fn drop(&mut self) {
+        unsafe {
+            if CHECK != self.0 - 1 {
+                panic!("Found {}, should have found {}", CHECK, self.0 - 1);
+            }
+            CHECK = self.0;
+        }
+    }
+}
+
+struct DropPanic;
+
+impl Drop for DropPanic {
+    fn drop(&mut self) {
+        panic!()
+    }
+}
+
+fn a() {
```

Comment:
Please duplicate `a` and `b` with versions that use `[DropChecker(1); N]` and that use `[DROP_CHECKER; N]` where `const DROP_CHECKER: DropChecker = DropChecker(1);`